service-registry/src/main/java/org/apache/servicecomb/serviceregistry/registry/RemoteServiceRegistry.java

```diff
@@ -50,18 +52,16 @@ public RemoteServiceRegistry(EventBus eventBus, ServiceRegistryConfig serviceReg
   public void init() {
     super.init();
     taskPool = new ScheduledThreadPoolExecutor(3,
-        task -> {
-          return new Thread(task) {
-            @Override
-            public void run() {
-              try {
-                setName("Service Center Task [" + task.toString() + "[" + this.getId() + "]]");
-                super.run();
-              } catch (Throwable e) {
-                LOGGER.error("task {} execute error.", getName(), e);
-              }
-            }
-          };
+        new ThreadFactory() {
+          private int taskId;
```

Comment:
It's better to assign a default value explicitly

Reply:
Done. pls check it.